test/expected/partitioning.out

```diff
@@ -369,3 +369,110 @@ ALTER SCHEMA my_partitioning_schema RENAME TO new_partitioning_schema;
 INSERT INTO part_custom_dim(time, combo) VALUES (now(), (3,4));
 NOTICE:  custom hash value is: 7
 NOTICE:  custom hash value is: 7
+-- Test partitioning function on an open (time) dimension
+CREATE OR REPLACE FUNCTION time_partfunc(unixtime float8)
+    RETURNS TIMESTAMPTZ LANGUAGE PLPGSQL IMMUTABLE AS
+$BODY$
+DECLARE
+    retval TIMESTAMPTZ;
+BEGIN
+
+    retval := to_timestamp(unixtime);
+    RAISE NOTICE 'time value for % is %', unixtime, timezone('UTC', retval);
+    RETURN retval;
+END
+$BODY$;
+CREATE TABLE part_time_func(time float8, temp float8, device text);
+\set ON_ERROR_STOP 0
+-- Should fail due to invalid time column
+SELECT create_hypertable('part_time_func', 'time');
+ERROR:  invalid dimension type: "time" must be an integer, date or timestamp
+\set ON_ERROR_STOP 1
+-- Should work with time partitioning function that returns a valid time type
+SELECT create_hypertable('part_time_func', 'time', time_func => 'time_partfunc');
+NOTICE:  adding not-null constraint to column "time"
+      create_hypertable       
+------------------------------
+ (17,public,part_time_func,t)
+(1 row)
+
+INSERT INTO part_time_func VALUES (1530214157.134, 23.4, 'dev1'),
+                                  (1533214157.8734, 22.3, 'dev7');
+NOTICE:  time value for 1530214157.134 is Thu Jun 28 19:29:17.134 2018
+NOTICE:  time value for 1530214157.134 is Thu Jun 28 19:29:17.134 2018
+NOTICE:  time value for 1530214157.134 is Thu Jun 28 19:29:17.134 2018
+NOTICE:  time value for 1530214157.134 is Thu Jun 28 19:29:17.134 2018
+NOTICE:  time value for 1533214157.8734 is Thu Aug 02 12:49:17.8734 2018
+NOTICE:  time value for 1533214157.8734 is Thu Aug 02 12:49:17.8734 2018
+NOTICE:  time value for 1533214157.8734 is Thu Aug 02 12:49:17.8734 2018
+NOTICE:  time value for 1533214157.8734 is Thu Aug 02 12:49:17.8734 2018
+SELECT time, temp, device FROM part_time_func;
+      time       | temp | device 
+-----------------+------+--------
+  1530214157.134 | 23.4 | dev1
+ 1533214157.8734 | 22.3 | dev7
+(2 rows)
+
+SELECT time_partfunc(time) at time zone 'UTC', temp, device FROM part_time_func;
+NOTICE:  time value for 1530214157.134 is Thu Jun 28 19:29:17.134 2018
+NOTICE:  time value for 1533214157.8734 is Thu Aug 02 12:49:17.8734 2018
+           timezone            | temp | device 
+-------------------------------+------+--------
+ Thu Jun 28 19:29:17.134 2018  | 23.4 | dev1
+ Thu Aug 02 12:49:17.8734 2018 | 22.3 | dev7
+(2 rows)
+
+SELECT * FROM test.show_subtables('part_time_func');
+                  Child                   | Tablespace 
+------------------------------------------+------------
+ _timescaledb_internal._hyper_17_11_chunk | 
+ _timescaledb_internal._hyper_17_12_chunk | 
+(2 rows)
+
+SELECT * FROM test.show_constraintsp('_timescaledb_internal._hyper_13%');
```

Comment:
shouldn't we see some constraints here?

Reply:
Yes there should be. Rebase changed table names, but should be fixed now.